### ISSUE TITLE
Merging rules

### ISSUE THREAD

**Issue by metalstrolch** [navit-gps/navit#383]
While having strong requirements for something being able to be merged into navit's trunk is generally a good idea, I encountered the question if enforcing a branch to be up to date with trunk before merging plus the requirement to have all checks passed in combination is not too much.

This makes working with multiple pull requests in parallel a really time consuming job. Since the merge of one pull request forces the updating of the others which forces rerunning all ci tests which seem to be somehow unreliable at the moment too. Think about coping for all those new icon things.

So I would prefer returning to old behaviour where it was not explicitly required for a branch to be up to date when merging.

What do you think?

**Comment by jkoan** [comment]
I would suggest that we should keep the behavior as it is now, except for images/icons and documentation. So probably because github cant handle this we need to disable the rule on github but need to respect it in our mind.

**Comment by sleske** [comment]
I'm also for "disable on github, respect in our mind". If you are certain the problems shown in the Github interface are unjustified, don't let them stop you.
Of course, I usually commit directly, not via a PR, so who I am to argue...

**Comment by metalstrolch** [comment]
My main reason to bring this on topic was mainly the reason that at the moment github refuses to merge even if there are no conflicts, just because the latest trunk was not merged to the branch. Which is quite annoying, since after every push to trunk you have to redo this, including otherwise unnecessary ci run.

So I vote for disabling that rule as well.

**Comment by sleske** [comment]
As I wrote in #389 , I think the present rules are  too restrictive.
Feature branches (which PRs are) should be restricted to risky, disruptive changes, or if a discussion is desired. Having a PR for every trivial change just feels wrong.

@pgrandin explained that one goal of the setting was to avoid breaking the main branch. That is a good point - however, that is why we have both "trunk" and "master". trunk is where development happens, and may break occasionally. trunk can be merged to master (or rather, master fast-forwarded to trunk) periodically once trunk has passed all CI tests.

If there are no objections, I'll disable the branch protection for trunk, so pushing and merging is unrestricted. We have worked like this for a long time, and I don't recall any drastic problems.

Comments? Especially from @pgrandin ?

**Comment by jkoan** [comment]
@sleske what about disableing the branch protection for trunk? Any progress? :D

**Comment by sleske** [comment]
@jkoan : Sorry this took so long, was busy for a while

Branch protection for trunk has been reduced - merging & pushing are now unrestricted.

The branch is still marked as "protected", which means that force-pushing and deletion are forbidden. That seems reasonable to me, so I left it as-is.

The settings for master are unchanged, and still require passing CircleCI checks before merge - so master is our "stable" branch (hopefully). I hope that works for everyone.
